docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,75 @@
+---
+keywords: [beginner, tutorial, compile]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's canisters, you need to compile the code into a WebAssembly module before it can be deployed on ICP. To build your canister's Wasm module, you can use the `dfx build` command. The `dfx build` command looks for source code to compile for each canister configured under the `canisters` section in the `dfx.json` configuration file. It can be used to build a specific canister or all canisters defined in the project's [`dfx.json`](/docs/current/developer-docs/developer-tools/cli-tools/dfx-json-reference) file.
```

Comment:
My high-level comment is that this page reads like a `dfx` reference for the `build` and `create` commands. Would it make sense to explain compilation on the conceptual level? I.e. how code is actually compiled to Wasm and Wasm targets we support? We mention that "code gets compiled to Wasm module" but we don't explain what that means. In other words, after reading this page they understand which `dfx` commands to invoke, but they still don't know how compilation is done conceptually.

Reply:
That's good feedback, thank you. I'll expand revise this to expand on the conceptual components.

Reply:
I've added some info about the conceptual information. Can you take another look?